sql: TestRelocateNonVoters failed

Failed in [this CI build](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/11168783?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true) with the following error:

```
Failed
=== RUN   TestRelocateNonVoters/ALTER_RANGE_x_RELOCATE_NONVOTERS
    test_server_shim.go:98:
        Test server was configured to route SQL queries to a secondary tenant (virtual cluster).
        If you are only seeing a test failure when this message appears, there may be a problem
        specific to cluster virtualization or multi-tenancy.
        To investigate, consider using "COCKROACH_TEST_TENANT=true" to force-enable just
        the secondary tenant in all runs (or, alternatively, "false" to force-disable), or use
        "COCKROACH_INTERNAL_DISABLE_METAMORPHIC_TESTING=false" to disable all random test variables altogether.
    testcluster.go:416:
        Test server was configured to route SQL queries to a secondary tenant (virtual cluster).
        If you are only seeing a test failure when this message appears, there may be a problem
        specific to cluster virtualization or multi-tenancy.
        To investigate, consider using "COCKROACH_TEST_TENANT=true" to force-enable just
        the secondary tenant in all runs (or, alternatively, "false" to force-disable), or use
        "COCKROACH_INTERNAL_DISABLE_METAMORPHIC_TESTING=false" to disable all random test variables altogether.
    multitenant_admin_function_test.go:369: condition failed to evaluate within 3m45s: from multitenant_admin_function_test.go:392: expected "trying to add a non-voter to a store that already has a NON_VOTER" contains "ok" tenant=secondary query=`ALTER RANGE (SELECT min(range_id) FROM [SHOW RANGES FROM TABLE t]) RELOCATE NONVOTERS FROM 2 TO 3;` leaseholder=1 replicas=[1 2 4 5] voting_replicas=[1 4 5] non_voting_replicas=[2] fromReplica=2 toReplica=3 row=0 col=2
    --- FAIL: TestRelocateNonVoters/ALTER_RANGE_x_RELOCATE_NONVOTERS (433.62s)
```

Jira issue: CRDB-30305

--- a/pkg/sql/create_as_test.go
+++ b/pkg/sql/create_as_test.go
@@ -325,6 +325,7 @@ AND status != 'succeeded'`
 
 func TestFormat(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	testCases := []struct {
 		sql            string

--- a/pkg/sql/create_function_test.go
+++ b/pkg/sql/create_function_test.go
@@ -151,6 +151,7 @@ SELECT nextval(105:::REGCLASS);`,
 
 func TestVersionGatingUDFInCheckConstraints(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	t.Run("new_schema_changer_version_enabled", func(t *testing.T) {
 		params, _ := createTestServerParams()

--- a/pkg/sql/multitenant_admin_function_test.go
+++ b/pkg/sql/multitenant_admin_function_test.go
@@ -246,7 +246,7 @@ func (tc testCase) runTest(
 	if numNodes == 0 {
 		numNodes = 1
 	}
-	cfg.ServerArgs.DefaultTestTenant = base.TestTenantProbabilistic
+	cfg.ServerArgs.DefaultTestTenant = base.TestControlsTenantsExplicitly
 	testCluster := serverutils.StartNewTestCluster(t, numNodes, cfg.TestClusterArgs)
 	defer testCluster.Stopper().Stop(ctx)
 

--- a/pkg/sql/schema_changer_test.go
+++ b/pkg/sql/schema_changer_test.go
@@ -7586,6 +7586,7 @@ CREATE TABLE t.test (pk INT PRIMARY KEY, v INT);
 // to conclude. If the locks were not dropped, a deadlock could occur.
 func TestConcurrentSchemaChangesDoNotDeadlock(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()
 	s, sqlDB, _ := serverutils.StartServer(t, base.TestServerArgs{})

--- a/pkg/sql/sequence_test.go
+++ b/pkg/sql/sequence_test.go
@@ -115,6 +115,7 @@ func BenchmarkUniqueRowID(b *testing.B) {
 // - ownership removal
 func TestSequenceOwnershipDependencies(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()
 	params := base.TestServerArgs{}

